src/RadhydroPulseMGint/test_radhydro_pulse_MG_int.cpp

```diff
@@ -0,0 +1,551 @@
+/// \file test_radhydro_pulse_MG_int.cpp
+/// \brief Defines a test problem for multigroup radiation in the diffusion regime with advection by gas using group-interpolated variable opacity.
+///
+
+#include "test_radhydro_pulse_MG_int.hpp"
+#include "AMReX_BC_TYPES.H"
+#include "AMReX_Print.H"
+#include "RadhydroSimulation.hpp"
+#include "fextract.hpp"
+#include "physics_info.hpp"
+#include "planck_integral.hpp"
+#include "radiation_system.hpp"
+
+struct MGintProblem {
+};		     // dummy type to allow compile-type polymorphism via template specialization
+struct GreyProblem { // Advecting grey pulse
+};
+
+AMREX_GPU_MANAGED double spec_power = -1.0; // NOLINT
+static constexpr bool export_csv = true;
+
+// constexpr int n_groups_ = 2;
+// constexpr amrex::GpuArray<double, n_groups_ + 1> rad_boundaries_{1e16, 1e18, 1e20};
+constexpr int n_groups_ = 4;
+constexpr amrex::GpuArray<double, n_groups_ + 1> rad_boundaries_{1e16, 1e17, 1e18, 1e19, 1e20};
```

Comment:
A design choice: it looks here like you're requiring the groups to be set at compile time. Do we want to keep that, or have the number of groups / group boundaries be settable at run-time, via the problem file? I guess I would favor setting things at run-time in the same way that we set the number of cells in the spatial dimensions at run-time, unless there are strong code speed / efficiency reasons to set them at compile time.

Reply:
I thought about that and found there are difficulties in reading rad_boundaries from the parameter file. The main difficulty is that many functions take `GpuArray(double, nGroups_)` as an input, so if nGroups_ has to read from the parameter file, then you have to replace GpuArray with a dynamic memory array like Vector. A lot of things have to be rewritten. I'll try to make that work in a future PR.

Reply:
OK, happy to leave this as set at compile-time for now.